src/cltl/entity_linking/linkers.py

```diff
@@ -42,35 +42,30 @@ def link(self, capsule):
         return capsule
 
     def link_entities(self, capsule):
-        if 'person' in capsule['subject']['type']:
-            uri = self._entity_search.search_entities_by_label(capsule['subject']['label'])
-            if uri:
-                capsule['subject']['uri'] = uri
-            else:
-                uri = self.fuzzy_label_match(capsule['subject']['label'])
-            if uri:
-                capsule['subject']['uri'] = uri
-            else:
-                capsule['subject']['uri'] = str(
-                    self._rdf_builder.create_resource_uri('LW', capsule['subject']['label'].lower()))
-        else:
-            capsule['subject']['uri'] = str(
-                self._rdf_builder.create_resource_uri('LW', capsule['subject']['label'].lower()))
+        capsule = self._link_entity(capsule, 'subject')
+        capsule = self._link_entity(capsule, 'object')
+        capsule = self._link_entity(capsule, 'author')
+
+        return capsule
+
+    def _link_entity(self, capsule, entity_position):
+        if entity_position not in capsule:
+            return capsule
 
-        if 'person' in capsule['object']['type']:
-            uri = self._entity_search.search_entities_by_label(capsule['object']['label'])
+        if 'person' in capsule[entity_position]['type']:
+            uri = self._entity_search.search_entities_by_label(capsule[entity_position]['label'])
             if uri:
-                capsule['object']['uri'] = uri
+                capsule[entity_position]['uri'] = uri
             else:
-                uri = self.fuzzy_label_match(capsule['object']['label'])
+                uri = self.fuzzy_label_match(capsule[entity_position]['label'])
             if uri:
-                capsule['object']['uri'] = uri
+                capsule[entity_position]['uri'] = uri
             else:
-                capsule['object']['uri'] = str(
-                    self._rdf_builder.create_resource_uri('LW', capsule['object']['label'].lower()))
+                capsule[entity_position]['uri'] = str(
+                    self._rdf_builder.create_resource_uri('LW', capsule[entity_position]['label'].lower()))
```

Comment:
And a question about the URIs we use, they are now based on the label, couldn't that lead to conflicts if e.g. two persons have the same name? Eventually we could use uuids instead.

Reply:
Yes, how we name the uri's is something that we need to think about. I was also thinking of adding an extra digit to each of the uri's (e.g. jaap-1, jaap-2, jaap-3) but that means keeping track of where we are. I think uuids would work better indeed.

Reply:
I combined the label with a timestamp to make it unique in the first chatbot version that differentiated identities. In that way you are sure the URI was not used before.